core/src/main/java/nl/nn/adapterframework/metrics/LocalDistributionSummary.java

```diff
@@ -0,0 +1,163 @@
+
+import io.micrometer.core.instrument.AbstractDistributionSummary;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.distribution.CountAtBucket;
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
+import io.micrometer.core.instrument.distribution.Histogram;
+import io.micrometer.core.instrument.distribution.HistogramSnapshot;
+import io.micrometer.core.instrument.distribution.TimeWindowFixedBoundaryHistogram;
+import io.micrometer.core.instrument.distribution.TimeWindowMax;
+
+public class LocalDistributionSummary extends AbstractDistributionSummary {
+	private static final CountAtBucket[] EMPTY_HISTOGRAM = new CountAtBucket[0];
+	private final LongAdder count;
+	private final DoubleAdder amount;
+	private final AtomicLong min;
+	private final TimeWindowMax max;
+	private final LongAdder sumOfSquares;
+	private final AtomicLong first;
+	private final AtomicLong last;
+	private final Histogram histogram; //Overwrite the default time bound histogram
+
+	public LocalDistributionSummary(Id id, Clock clock, DistributionStatisticConfig distributionStatisticConfig, double scale, boolean supportsAggregablePercentiles) {
+		super(id, clock, distributionStatisticConfig, scale, supportsAggregablePercentiles);
+
+		histogram = new TimeWindowFixedBoundaryHistogram(clock, DistributionStatisticConfig.builder()
+				.expiry(Duration.ofDays(1825)) // effectively never roll over
+				.bufferLength(1)
+				.build()
+				.merge(distributionStatisticConfig), true);
+
+		this.min = new AtomicLong(Long.MAX_VALUE);
+		this.sumOfSquares = new LongAdder();
+		this.first = new AtomicLong(-1);
+		this.last = new AtomicLong();
+		this.count = new LongAdder();
+		this.amount = new DoubleAdder();
+		this.max = new TimeWindowMax(clock, distributionStatisticConfig);
+	}
+
+	@Override
+	protected void recordNonNegative(double amount) {
+		checkMinMax((long) amount);
+		addSums((long) amount);
+		updateFirstLast((long) amount);
+
+		count.increment();
+		this.amount.add(amount);
+		max.record(amount);
+
+		if(histogram != null) {
+			histogram.recordDouble(amount);
+		}
+	}
+
+	@Override
+	public long count() {
+		return count.longValue();
+	}
+
+	@Override
+	public double totalAmount() {
+		return amount.doubleValue();
+	}
+
+	@Override
+	public double max() {
+		return max.poll();
+	}
+
+	private void updateFirstLast(long value) {
+		first.compareAndSet(-1, value);
+		last.set(value);
+	}
+
+	public void checkMinMax(long value) {
+		if (value < min.get()) {
+			min.set(value);
+		}
+	}
+
+	protected void addSums(long value) {
+		sumOfSquares.add(value * value);
+	}
+
+	public double getVariance() {
+		return calculateVariance(count(), Math.round(totalAmount()), sumOfSquares.sum());
+	}
+	private double calculateVariance(long count, long sum, long sumOfSquares) {
+		if (count>1) {
+			return (sumOfSquares-((sum*sum)/count))/(count-1);
+		} else {
+			return Double.NaN;
+		}
+	}
```

Comment:
Mijn 'Basics' werkte met integers als te meten waarden, en gebruikte daarom long in calculateVariance. Als recordNonNegative()  nu met double werkt, moeten calculateVariance, addSums, checkMinMax, checkFirstLast dat ook gaan doen.